plugins/catalog-backend-module-github/src/analyzers/GithubLocationAnalyzer.ts

```diff
@@ -77,7 +77,7 @@ export class GithubLocationAnalyzer implements ScmLocationAnalyzer {
 
     const catalogFile = catalogFilename || 'catalog-info.yaml';
 
-    const query = `filename:${catalogFile} repo:${owner}/${repo}`;
+    const query = `filename:${catalogFile} extension:yaml OR extension:yml repo:${owner}/${repo}`;
```

Comment:
I assume the `filename:` matching is partial then? i.e. `filename:foo` matches `foobar`.

Is there a way to solve that instead, so that only files with an exact match are included? We don't necessarily know that all files will use the `yaml` or `yml` extension since you can define your own parser.

Reply:
Yes the other option I tried was using a regex - sounds like thats better if there are other extensions possible.

Reply:
Well I couldn't get regex working actually so I just extracted the extension dynamically.